pkg/evmclient/client.go

```diff
@@ -293,6 +295,386 @@ func (c *Client) FilterLogs(ctx context.Context, q FilterQuery) ([]types.Log, er
 	}
 }
 
+func (c *Client) GetFullBlockByHash(ctx context.Context, hash string) (*pb.Block, error) {
```

Comment:
Should not return a protobuf type here.

Reply:
Do you mean, we should not use protobuf as internal struct usage? 
I think it can be used as a Golang struct to pass around, isn't it?
Do you have any suggestions for this?

Reply:
We often use protobuf types for communicating between services, it's should not be use for internal data between components of the service.